### PR TITLE
Finish Windows service installer

### PR DESCRIPTION
**Pull request by samkearney**
- Harmonize MajorUpgrade usage in the service installer MSI with the method used for our other products
- Switch to using Wix ServiceInstall in the installer and merge module, so that the service will be refcounted properly.

### CODE REVIEW

File: src/windows/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+# service_config.h, are defined below.
+
+set(BROKER_SERVICE_NAME "ETC RDMnet Broker")
+set(BROKER_SERVICE_DISPLAY_NAME "ETC RDMnet Broker")
+set(BROKER_SERVICE_DESCRIPTION "Implements an RDMnet message broker as defined in ANSI E1.33")
+
+# The following pairs must match; see https://wixtoolset.org/documentation/manual/v3/xsd/wix/serviceinstall.html
+# and https://docs.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-createservicew
+set(BROKER_SERVICE_C_START_TYPE SERVICE_AUTO_START)
+set(BROKER_SERVICE_WIX_START_TYPE "auto")
```
Comment (bhair) [inline src/windows/CMakeLists.txt:18]: So are you just having cmake create these are variables that both files can then consume? That seems pretty great.

File: tools/install/windows/merge/BrokerMerge.wxs [hunk 2]
```diff
@@ -17,7 +18,7 @@
       <Directory Id="$(var.ProgramFilesFolder)">
         <Directory Id="PROGRAM_FILES_ETC_DIR" Name="ETC">
           <Directory Id="INSTALLDIR" Name="RDMnetBroker" FileSource="$(var.InstallSourceDir)">
-            <Component Id="Broker" Guid="2B4014E4-5278-469A-ABA7-A94F0AB850DA" SharedDllRefCount='yes'>
+            <Component Id="Broker" Guid="2B4014E4-5278-469A-ABA7-A94F0AB850DA" SharedDllRefCount="yes">
```
Comment (bhair) [inline tools/install/windows/merge/BrokerMerge.wxs:21]: Is this explicit mention of the sharedDllRefCount required in addition to the Service elements for the shareddllrefcounts to work correctly?

File: tools/install/windows/merge/BrokerMerge.wxs [hunk 3]
```diff
@@ -26,8 +27,9 @@
                     Profile="all" Protocol="udp" Scope="any" IgnoreFailure="no" />
               </File>
               <File Id="DnsSdDll" Name="dnssd.dll" DiskId="1" />
-              <ServiceControl Id="BrokerServiceCtl" Name="ETC RDMnet Broker" Stop="both"
-                  Start="install" />
+
+              <!-- Configures the actual installing and uninstalling of the service -->
+              <?include GeneratedFiles\service_config.wxi ?>
```
Comment (bhair) [inline tools/install/windows/merge/BrokerMerge.wxs:32]: Was having cmake generate a file with the values required a cleaner solution than just having cmake pass these in as variables to the wix compiler as cmd line arguments?

Comment (nwagner) [reply tools/install/windows/merge/BrokerMerge.wxs:32]: I would say it depends on the number of arguments we would have to pass.  One or two is fine, but more than that a separate include is easier to maintain.

**Review by nwagner — APPROVED** [review]

File: src/windows/CMakeLists.txt (hunk 1 above)
Comment (samkearney) [reply src/windows/CMakeLists.txt:18]: Yes, the variables are referenced by the `configure_file()` statements below.

File: tools/install/windows/merge/BrokerMerge.wxs (hunk 3 above)
Comment (samkearney) [reply tools/install/windows/merge/BrokerMerge.wxs:32]: Agree with @nwagner , have a look at src/windows/service_config.wxi.in; every value enclosed in "@" symbols would need to be passed at the command line. Plus, in my experience it's easier to reason about software if all or at least most of your definitions are in a source file -- hunting into the build tools and compiler invocation commands to find where something is defined makes the software more difficult to understand. Using the file is cleaner IMO.

File: tools/install/windows/merge/BrokerMerge.wxs (hunk 2 above)
Comment (samkearney) [reply tools/install/windows/merge/BrokerMerge.wxs:21]: I don't know the answer to that; I was under the impression (I think you told me) that you needed that attribute in order for the component to be properly refcounted.

File: tools/install/windows/merge/BrokerMerge.wxs (hunk 3 above)
Comment (bhair) [reply tools/install/windows/merge/BrokerMerge.wxs:32]: That makes perfect reasonable sense. Just didn't know if there was a reason beyond ease of maintenance.

File: tools/install/windows/merge/BrokerMerge.wxs (hunk 2 above)
Comment (bhair) [reply tools/install/windows/merge/BrokerMerge.wxs:21]: For a normal file, yes that's required. I was just wondering if the service control element required that to function or if the service elements would handle the refcounting on their own.

Comment (samkearney) [reply tools/install/windows/merge/BrokerMerge.wxs:21]: Well I just combined the ServiceControl and File elements under the same component, for convenience, so I haven't tested the alternative.

**Pull request merged**